src/SoftwareSerial.h

```diff
@@ -32,11 +32,76 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 // If only one tx or rx wanted then use this as parameter for the unused pin
 constexpr int SW_SERIAL_UNUSED_PIN = -1;
 
-enum SoftwareSerialConfig {
-	SWSERIAL_5N1 = 0,
-	SWSERIAL_6N1,
-	SWSERIAL_7N1,
-	SWSERIAL_8N1,
+// Options for SoftwareSerialConfig
+constexpr uint8_t SWSERIAL_NB_BIT_MASK       = 0B00011000;
+constexpr uint8_t SWSERIAL_NB_BIT_5          = 0B00000000;
+constexpr uint8_t SWSERIAL_NB_BIT_6          = 0B00001000;
+constexpr uint8_t SWSERIAL_NB_BIT_7          = 0B00010000;
+constexpr uint8_t SWSERIAL_NB_BIT_8          = 0B00011000;
+
+constexpr uint8_t SWSERIAL_PARITY_CONF_MASK  = 0B00000111;
+constexpr uint8_t SWSERIAL_PARITY_CONF_NONE  = 0B00000000;
+constexpr uint8_t SWSERIAL_PARITY_CONF_EVEN  = 0B00000010;
+constexpr uint8_t SWSERIAL_PARITY_CONF_ODD   = 0B00000011;
+constexpr uint8_t SWSERIAL_PARITY_CONF_MARK  = 0B00000100;
+constexpr uint8_t SWSERIAL_PARITY_CONF_SPACE = 0B00000101;
+
+constexpr uint8_t SWSERIAL_NB_STOP_BIT_MASK  = 0B01100000;
+constexpr uint8_t SWSERIAL_NB_STOP_BIT_0     = 0B00000000;
+constexpr uint8_t SWSERIAL_NB_STOP_BIT_1     = 0B00100000;
+constexpr uint8_t SWSERIAL_NB_STOP_BIT_15    = 0B01000000;
```

Comment:
is there prior art for this 1.5 thing?

Reply:
I have no idea how common it is in real life. Mainly used as 5N1.5 by old hardware such as [teletypes](https://en.wikipedia.org/wiki/Teleprinter), which need the time to reset or process the data. Strangely enough it is still supported by most hardware UARTs. It also [seems to be used](https://www.raspberrypi.org/forums/viewtopic.php?t=207600) by some devices at 9600, 8E1.5. The proper way to do it is to use a stop bit length of 1.5 m_bitCycles. The cheap way is to use 2 stop bits for tx and 1 stop bit for rx.

While 2 stop bits may seem even less useful, it is probably more common in real life. E.g. in ModBus as 8N2 (which is equvialent to 8M1) and by the [DMX512 protocol](https://erg.abdn.ac.uk/users/gorry/eg3576/DMX-frame.html) used for stage ligthing, albeit at 250 kbaud.

Even if need is limited for these variants, I think it would be interesting to allow the flexibility if it can be supported with minimal overhead.

On a second thought, 2 stop bits may not be ideally implemented now. You really only need to detect the first stop bit and then reset and wait for the next start bit. On the other hand, requiring both stop bits might help to detect frame errors. The costs/benefits should preferably be considered in more detail.